### ISSUE TITLE
Support copy from stdin

### ISSUE THREAD

**Issue by jran-dasa** [intel/bmap-tools#74]
I found an old issue from Tizen, that was closed since there were supposedly "no real use-cases".
https://bugs.tizen.org/browse/DEVT-142
I found this since I do have a real use-case, and tried using stdin (simply specifying "-") as input without any luck.

In our application, We're trying to use bmaptool for factory programming of an embedded device.
This is done using NXP tool "UUU", which can send files via USB to stdin. If this file could be piped to bmaptool it'd all work just fine.
I cannot download the entire image file to the device, as it is to large to fit into RAM. And the flash memory is of course to be used as target memory.
dd handles this just fine, but we'd of course prefer the speed gain that comes with bmaptool's sparseness handling.

Would it be possible to have bmaptool supporting input from stdin?

**Comment by dedekind** [comment]
Hi, I am not actively working on this tool anymore, and do not really have bandwidth to work on this for you. In the past I my company sponsored me to work on bmap-tools and Tizen stuff, but not anymore.

But sure if you or someone is willing to implement it, pull requests are welcome. I can also help by reviewing and I can participate in design/planning discussion if needed. Thanks!

**Comment by jpfaff** [comment]
Hi @jran-dasa,
Have you found a way of passing an image with bmaptool to "UUU"?

**Comment by jran-dasa** [comment]
> Hi @jran-dasa, Have you found a way of passing an image with bmaptool to "UUU"?

I did not. Can't quite remember the exact details, but some how I managed to get the entire image (compressed into a wic.gz file) to fit in a RAM disk. So programming takes some extra time with the extra step of first copying the entire file into RAM before starting to write it into flash, but it is still a whole lot faster then using dd to write the entire flash drive.
